### PR TITLE
Modify example apps to use json config instead of direct parameters

### PR DESCRIPTION
**Pull request by alexr-bq**
Should not be merged until after release of 12.0 driver

### CODE REVIEW

**Comment by ansleis** [comment]
I think the ODBC readme has to be updated to specify the 12.0 driver.

File: JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java [hunk 1]
```diff
@@ -61,10 +61,15 @@ private static Connection connectToDB(String host, String port, String dbName, S
...
-        jdbcOptions.put("oauthclientid", clientId);
-        jdbcOptions.put("oauthclientsecret", clientSecret);
-	jdbcOptions.put("oauthscope", scope);
+
+	// Put these options into static json config
+	String jsonConfig = "{\"oauthtokenurl\" : \"" + tokenUrl + "\", " +  
+	"\"oauthclientid\" : \"" + clientId + "\", " + 
+	"\"oauthclientsecret\" : \"" + clientSecret + "\", " +
+	"\"oauthscope\" : \"" + scope + "\"" +
+	"}";
```
Comment (jeremyp-bq) [inline JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java:70]: This makes it harder to read than before.  Too bad Python-like multi-line strings are only available in Java 15+.

Do we have to make this change for these examples (i.e. do we recommended always using the JSON for some of the OAuth params)?

Comment (alexr-bq) [reply JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java:70]: Yes, the change in functionality is to use json configuration for static configuration, and only maintain the parameters for that configuration for backwards compat.

**Comment by alexr-bq** [comment]
> I think the ODBC readme has to be updated to specify the 12.0 driver.

Added

**Comment by ansleis** [comment]
Apparently github doesn't let you comment on collapsed lines in the diff, but I think this should be changed also:

In both the JDBC and ODBC readmes:

Currently: 
The server must be configured for OAuth as described [here](https://www.vertica.com/docs/latest/HTML/Content/Authoring/Security/ClientAuth/OAuth/OAuthODBC.htm)

Change to:
The server must be configured for OAuth as described in the [Vertica documentation](https://www.vertica.com/docs/latest/HTML/Content/Authoring/Security/ClientAuth/OAuth/ConfiguringOAuth.htm).

This link isn't currently active, but it will be after I publish for GA.

**Review by williammatherjones — APPROVED** [review]

**Pull request merged**